src/main/java/com/stocks/project/controller/UserController.java

```diff
@@ -98,37 +95,33 @@ public ResponseEntity<?> deleteUser(@PathVariable int userId) {
     }
 
     @PutMapping("/{userId}")
-    public ResponseEntity<?> updateUser(@RequestBody User updatedUser, @PathVariable int userId) {
-        try {
-            return new ResponseEntity<>(userService.updateUser(updatedUser, userId), HttpStatus.OK);
-        } catch (NoSuchUserException e) {
-            return ResponseEntity
-                    .status(HttpStatus.NOT_FOUND)
-                    .contentType(MediaType.APPLICATION_JSON)
-                    .body("""
-                    {
-                        "error" : "No such user"\s
-                    }
-                    """);
-        }
-    }
-
-    @PostMapping("/register")
-    public ResponseEntity<?> registerUser(@RequestBody UserSecurityDTO dto) {
-        try {
-            return new ResponseEntity<>(userService.register(dto), HttpStatus.OK);
-        } catch (Exception e) {
-            return ResponseEntity
-                    .status(HttpStatus.BAD_REQUEST)
-                    .contentType(MediaType.APPLICATION_JSON)
-                    .body("""
-                            
-                            """);
+    public ResponseEntity<?> updateUser(@RequestBody User updatedUser,
+                                        @PathVariable int userId,
+                                        Principal principal) {
+        String login = principal.getName();
+        if (userService.isAdmin(login)) {
```

Comment:
can we use here if (userService.isAdmin(login) || userService.isSame(login, userId)) ?